{
  "pageType":"tag",
  "title":"Throw",
  "name":"Throw",
  "description":"The cfthrow tag raises a developer-specified exception that can be caught with cfcatch tag\n  having any of the following type specifications - cfcatch type = 'custom_type', cfcatch type = 'Application'\n  'cfcatch' type = 'Any'",
  "status":null,
  "appendix":false,
  "bodyContentType":"empty",
  "attributeType":"fixed",
  "minimumAttributes":null,
  "handleException":null,
  "allowRemovingLiteral":null,
  "sourceClasses":{
    "tag":"lucee.runtime.tag.Throw",
    "ttt":null,
    "tdbt":null,
    "att":null,
    "tte":"lucee.transformer.cfml.evaluator.impl.Throw"
  },
  "script":{
    "type":"single",
    "runtimeExpressionValue":"true",
    "context":null
  },
  "attributes":[
    {
      "name":"type",
      "type":"string",
      "default":false,
      "defaultValue":null,
      "description":"A custom type or the predefined type Application. Do not enter any other predefined types because\n\t\tthey are not generated by CFML applications. If you specify the exception type Application, you\n\t\tneed not specify a type for cfcatch, because the Application type is the default cfcatch type",
      "aliases":[
        
      ],
      "values":[
        
      ],
      "status":null,
      "required":false,
      "noname":false,
      "scriptSupport":null,
      "runTimeExpressionValue":true
    },
    {
      "name":"message",
      "type":"any",
      "default":false,
      "defaultValue":null,
      "description":"A message that describes the exceptional event.",
      "aliases":[
        
      ],
      "values":[
        
      ],
      "status":null,
      "required":false,
      "noname":false,
      "scriptSupport":"optional",
      "runTimeExpressionValue":true
    },
    {
      "name":"detail",
      "type":"string",
      "default":false,
      "defaultValue":null,
      "description":"A detailed description of the event. The CFML server appends the position of the error to\n\t\tthis description; the server uses this parameter if an error is not caught by your code.",
      "aliases":[
        
      ],
      "values":[
        
      ],
      "status":null,
      "required":false,
      "noname":false,
      "scriptSupport":null,
      "runTimeExpressionValue":true
    },
    {
      "name":"errorCode",
      "type":"string",
      "default":false,
      "defaultValue":null,
      "description":"A custom error code that you supply.",
      "aliases":[
        
      ],
      "values":[
        
      ],
      "status":null,
      "required":false,
      "noname":false,
      "scriptSupport":null,
      "runTimeExpressionValue":true
    },
    {
      "name":"extendedInfo",
      "type":"string",
      "default":false,
      "defaultValue":null,
      "description":"extended information to the exception.",
      "aliases":[
        
      ],
      "values":[
        
      ],
      "status":null,
      "required":false,
      "noname":false,
      "scriptSupport":null,
      "runTimeExpressionValue":true
    },
    {
      "name":"object",
      "type":"any",
      "default":false,
      "defaultValue":null,
      "description":"a native java exception Object, if this attribute is defined all other will be ignored",
      "aliases":[
        
      ],
      "values":[
        
      ],
      "status":null,
      "required":false,
      "noname":false,
      "scriptSupport":null,
      "runTimeExpressionValue":true
    },
    {
      "name":"contextlevel",
      "type":"numeric",
      "default":false,
      "defaultValue":null,
      "description":"entry context level",
      "aliases":[
        
      ],
      "values":[
        
      ],
      "status":"hidden",
      "required":false,
      "noname":false,
      "scriptSupport":null,
      "runTimeExpressionValue":true
    }
  ],
  "examples":[
    
  ],
  "history":[
    
  ]
}